### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by sethgoodluck** [braposo/react-text-loop#41]
This component is amazing. 
However, I believe it may have a memory leak. 

![Screen Shot 2019-11-07 at 9 57 03 PM](https://user-images.githubusercontent.com/55430175/68453182-b2dab580-01a9-11ea-890e-b2824c36b28b.png)

How I'm using it: 

I created a react component as follows: 

```
import React from 'react';
import TextLoop from 'react-text-loop';

const BuiltForBlock = ({
	builtFor,
	A,
	B,
	C,
	D,
	E
}) => (
	<h2 className='title animated-main' style={{ color: builtFor.color }}>
		<span> {builtFor.text} </span>
		<TextLoop springConfig={{ stiffness: 80, damping: 10 }}>
			<span style={{ color: `${A.color}` }}>
				{A.text}
			</span>
			<span style={{ color: `${B.color}` }}>{B.text}</span>
			<span style={{ color: `${C.color}` }}>{C.text}</span>
			<span style={{ color: `${D.color}` }}>{D.text}</span>
			<span style={{ color: `${E.color}` }}>
				<strong>{E.text}</strong>
			</span>
		</TextLoop>
	</h2>
);



BuiltForBlock.defaultProps = {
	A: { color: 'red', text: 'A' },
	B: { color: 'blue', text: 'B' },
	C: { color: 'green', text: 'C' },
	D: { color: 'purple', text: 'D' },
	E: { color: 'orange', text: 'E' },
	builtFor: { color: 'white', text: 'Built For' }
};

export default BuiltForBlock;
```

Then within another component, I have this one. 

However, when switching pages on my react app, I believe it is unmounting mid-transition and not properly closing everything out? 

If it is a memory leak, that could be a real issue worth looking into and was hoping to get some insight on how to properly unmount it or call your attention to it to fix. 

Cheers :)

**Comment by zamson** [comment]
I have the same issue / warning in my app

**Comment by braposo** [comment]
@sethgoodluck @zamson Thanks for pointing this out! It's probably something related with tick timeouts, I'll have a look and get back to you.

**Comment by braposo** [comment]
@sethgoodluck @zamson it should be fixed if you update to v2.2.1

Can you please confirm it's all working in your code? Thanks!

**Comment by sethgoodluck** [comment]
That cleared it up for me alright! 
Thank you @braposo
